modular_skyrat/modules/modular_items/lewd_items/code/lewd_clothing/shibari_worn.dm

```diff
@@ -0,0 +1,330 @@
+
+
+/obj/item/clothing/under/shibari/proc/handle_take_off(datum/source, mob/user)
+	SIGNAL_HANDLER
+	INVOKE_ASYNC(src, .proc/handle_take_off_async, user)
+	return COMPONENT_CANCEL_ATTACK_CHAIN
+
+/obj/item/clothing/under/shibari/proc/handle_take_off_async(mob/user)
+	if(iscarbon(user))
+		var/mob/living/carbon/human/hooman = user
+		if(do_after(hooman, HAS_TRAIT(hooman, TRAIT_RIGGER) ? 2 SECONDS : 10 SECONDS, target = src))
+			qdel(src)
+
+/obj/item/clothing/under/shibari/ComponentInitialize()
+	. = ..()
+	AddElement(/datum/element/update_icon_updates_onmob)
+
+/obj/item/clothing/under/shibari/AltClick(mob/user)
+	. = ..()
+	var/mob/living/carbon/human/hooman = loc
+	if(user != hooman)
+		switch(tightness)
+			if(SHIBARI_TIGHTNESS_LOW)
+				tightness = SHIBARI_TIGHTNESS_MED
+			if(SHIBARI_TIGHTNESS_MED)
+				tightness = SHIBARI_TIGHTNESS_HIGH
+			if(SHIBARI_TIGHTNESS_HIGH)
+				tightness = SHIBARI_TIGHTNESS_LOW
+
+/obj/item/clothing/under/shibari/process(delta_time)
+	var/mob/living/carbon/human/hooman = loc
+	if(tightness == SHIBARI_TIGHTNESS_LOW && hooman.arousal < 15)
+		hooman.adjustArousal(0.6 * delta_time)
+	if(tightness == SHIBARI_TIGHTNESS_MED && hooman.arousal < 25)
+		hooman.adjustArousal(0.6 * delta_time)
+	if(tightness == SHIBARI_TIGHTNESS_HIGH && hooman.arousal < 30)
+		hooman.adjustArousal(0.6 * delta_time)
+
+//stuff to apply processing on equip and add mood event for perverts
+/obj/item/clothing/under/shibari/equipped(mob/user, slot)
+	.=..()
+	var/mob/living/carbon/human/hooman = user
+	if(src == hooman.w_uniform)
+		START_PROCESSING(SSobj, src)
+	if(HAS_TRAIT(hooman, TRAIT_ROPEBUNNY))
+		hooman.apply_status_effect(/datum/status_effect/ropebunny)
+
+//same stuff as above but for dropping item
+/obj/item/clothing/under/shibari/dropped(mob/user, slot)
+	.=..()
+	STOP_PROCESSING(SSobj, src)
+	var/mob/living/carbon/human/hooman = user
+	if(HAS_TRAIT(hooman, TRAIT_ROPEBUNNY))
+		hooman.remove_status_effect(/datum/status_effect/ropebunny)
+
+/obj/item/clothing/under/shibari/torso
+	name = "shibari ropes"
+	desc = "Nice looking rope bondage."
+	icon_state = "shibari_body"
+
+	greyscale_config = /datum/greyscale_config/shibari_clothes/body
+	greyscale_config_worn = /datum/greyscale_config/shibari_worn/body
+	greyscale_config_worn_digi = /datum/greyscale_config/shibari_worn_digi/body
+	greyscale_config_worn_taur_snake = /datum/greyscale_config/shibari_worn_taur_snake/body
+	greyscale_config_worn_taur_paw = /datum/greyscale_config/shibari_worn_taur_paw/body
+	greyscale_config_worn_taur_hoof = /datum/greyscale_config/shibari_worn_taur_hoof/body
+	greyscale_colors = "#bd8fcf"
+
+//processing stuff
+/obj/item/clothing/under/shibari/torso/process(delta_time)
+	. = ..()
+	var/mob/living/carbon/human/hooman = loc
+	if(tightness == SHIBARI_TIGHTNESS_HIGH && hooman.pain < 30)
+		if(hooman.pain < 25)
+			hooman.adjustPain(0.6 * delta_time)
+		if(prob(10))
+			hooman.adjustOxyLoss(5)
```

Comment:
No.

Reply:
Ok, we’ll fix it

Reply:
> No. Sex toys need to not touch any of the game balance or health stuff.

Then why do the large gags kill you from O2 deprivation?

Reply:
we need to remove all that too @chizzyyyy